app/ide-desktop/lib/client/src/config/parser.ts

```diff
@@ -0,0 +1,319 @@
+            if (match != null) {
+                const optionName = match[1] as string
+                const optionValue = match[3]
+                if (optionValue != null) {
+                    chromeOptions.push(new ChromeOption(optionName, optionValue))
+                } else {
+                    const nextArgValue = processArgs[i + 1]
+                    if (nextArgValue != null && !nextArgValue.startsWith('-')) {
+                        chromeOptions.push(new ChromeOption(optionName, nextArgValue))
+                        i++
+                    } else {
+                        chromeOptions.push(new ChromeOption(optionName, undefined))
+                    }
+                }
+            } else {
+                argv.push(processArg)
+            }
+        }
+    }
+    return { argv, chromeOptions }
+}
+
+// =====================
+// === Option Parser ===
+// =====================
+
+export function parseArgs() {
+    const args = config.config
+    const { argv, chromeOptions } = argvAndChromeOptions(fixArgvNoPrefix(hideBin(process.argv)))
+
+    const yargOptions = args.optionsRecursive().reduce((opts: { [key: string]: any }, option) => {
+        const yargsParam = Object.assign({}, option)
+        // @ts-ignore
+        yargsParam.requiresArg = ['string', 'array'].includes(yargsParam.type)
+        // @ts-ignore
+        yargsParam.default = undefined
+        // @ts-ignore
+        opts[naming.camelToKebabCase(option.qualifiedName())] = yargsParam
+        return opts
+    }, {})
+
+    let optParser = yargs()
+        .version(false)
+        .parserConfiguration({
+            // Allow single-dash arguments, like `-help`.
+            'short-option-groups': false,
+            // Treat dot-arguments as string keys, like `foo.bar`.
+            'dot-notation': false,
+            // Makes all flags passed after '--' be one string.
+            'populate--': true,
+            // Do not expand `--foo-bar` to `--fooBar`. This prevents an error when both the former
+            // and later argument are reported as invalid at the same time.
+            'camel-case-expansion': false,
+        })
+        .help(false)
+        .strict()
+        .options(yargOptions)
+
+    // === Parsing ===
+
+    let parseError = null as null | Error
+    let xargs = optParser.parse(argv, {}, (err: Error | undefined) => {
+        if (err != null) {
+            parseError = err
+        }
+    }) as { [key: string]: string }
+    const unexpectedArgs = xargs['--']
+
+    for (const option of args.optionsRecursive()) {
+        const arg = xargs[naming.camelToKebabCase(option.qualifiedName())]
+        if (arg != null) {
+            option.value = arg
+            option.setByUser = true
+        }
+    }
+
+    let windowSize = config.WindowSize.default()
+    const parsedWindowSize = config.WindowSize.parse(args.groups.window.options.size.value)
+
+    if (parsedWindowSize instanceof Error) {
+        throw 'wrong window size'
```

Comment:
Strings should not be thrown as errors, we should use `Error`. Also, message should provide some context (i.e. what the provided value was and why it is wrong).

Reply:
Or perhaps `WindowSize.parse` should throw rather than return error. That'd give cleaner code here.

Reply:
I don't like throws in libs code because TS type checker doesn't enforce you to handle these cases. That's why I prefer to return errors.